coordinator/internal/logic/provertask/chunk_prover_task.go

```diff
@@ -63,13 +66,24 @@ func (cp *ChunkProverTask) Assign(ctx *gin.Context, getTaskParameter *coordinato
 		return nil, fmt.Errorf("check prover task parameter failed, error:%w", err)
 	}
 
+	hardForkNumber, err := cp.getHardForkNumberByName(getTaskParameter.HardForkName)
+	if err != nil {
+		log.Error("hard fork get empty chunk because of the hard fork name don't exist", "fork name", getTaskParameter.HardForkName)
+		return nil, err
+	}
+
+	fromBlockNum, toBlockNum := forks.BlockRange(hardForkNumber, cp.forkHeights)
+	if toBlockNum > getTaskParameter.ProverHeight {
```

Comment:
what if fromBlockNum > ProverHeight?

Reply:
I also considered this. I think it's implicitly considered here: we'll select `height >= forkBlockNum && `height < proverHeight` which will be an empty range.

Reply:
Makes sense, might be beneficial to put an explicit check here as well. Up to @georgehao

Reply:
> we'll select height >= forkBlockNum && height < proverHeight` which will be an empty range.
yeah. the sql will check it. I think don't need other check.